client/src/featureform/register.py

```diff
@@ -3662,6 +3686,27 @@ def avg_user_transactions(client, params, entities):
         else:
             return decorator(fn)
 
+    def _register_project(
```

Comment:
In this file, I see two ways for a resource to be registered.
1. Using the `Registrar` class and appending to `self.__resources`
2. Make a separate class eg `Variants` that have a `register` method. 

What's the best strategy to support project registration among the above options?